gateway/mw_graphql.go

```diff
@@ -43,5 +45,19 @@ func (m *GraphQLMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Reques
 		return err, http.StatusBadRequest
 	}
 
+	result, err := gqlRequest.ValidateForSchema(m.Schema)
+	if err != nil {
+		m.Logger().Errorf("Error while validating GraphQL request: '%s'", err)
+		return errors.New("there was a problem proxying the request"), http.StatusInternalServerError
+	}
+
+	if result.Errors != nil && result.Errors.Count() > 0 {
+		w.Header().Set(headers.ContentType, headers.ApplicationJSON)
+		w.WriteHeader(http.StatusBadRequest)
+		_, _ = result.Errors.WriteResponse(w)
```

Comment:
is it good not to handle a write error here?

Reply:
Errors are built-in in library and they are tested,  not handling them in gateway makes code clear I think